pkg/sbom/io/decode.go

```diff
@@ -391,8 +391,19 @@ func (m *Decoder) addOrphanPkgs(sbom *types.SBOM) error {
 	for _, pkgs := range osPkgMap {
 		// TODO: mismatch between the OS and the packages should be rejected.
 		// e.g. OS: debian, Packages: rpm
-		sort.Sort(pkgs)
-		sbom.Packages = append(sbom.Packages, ftypes.PackageInfo{Packages: pkgs})
+
+		// Find existing PackageInfo with empty FilePath to merge with
+		if _, idx, found := lo.FindIndexOf(sbom.Packages, func(pkg ftypes.PackageInfo) bool {
+			return pkg.FilePath == ""
+		}); found {
+			// Merge with existing PackageInfo
+			sbom.Packages[idx].Packages = append(sbom.Packages[idx].Packages, pkgs...)
+			sort.Sort(sbom.Packages[idx].Packages)
+		} else {
+			// Create new PackageInfo
+			sort.Sort(pkgs)
+			sbom.Packages = append(sbom.Packages, ftypes.PackageInfo{Packages: pkgs})
+		}
```

Comment:
Do we need to use this difficalt logic?

I understand you idea - detected OS packages don't have filepath (i mean PackageInfo struct):
https://github.com/aquasecurity/trivy/blob/e76c7ee4efcad4a7b5c6e8e8a290a142dbb039bb/pkg/sbom/io/decode.go#L332-L338

But in `addOrphanPkgs` function `sbom.Packages` always contains 0 or 1 element.
And we check only first iteration of the `osPkgMap` loop.
https://github.com/aquasecurity/trivy/blob/e76c7ee4efcad4a7b5c6e8e8a290a142dbb039bb/pkg/sbom/io/decode.go#L408
It means that we don't need to check filePath and we can just add orphan packages into `sbom.Packages[0]`.

e.g.:
```suggestion
		if len(sbom.Packages) == 0 {
			sbom.Packages = []ftypes.PackageInfo{{}}
		}
		sbom.Packages[0].Packages = append(sbom.Packages[0].Packages, pkgs...)
		sort.Sort(sbom.Packages[0].Packages)
```

Reply:
There's a possibility that OS package file paths may be filled in the future, and I'm concerned that writing logic that's easily broken when other parts are changed could lead to bugs. However, as things stand now, you're right, so I simplified it.
https://github.com/aquasecurity/trivy/pull/9194/commits/1f0e04e4da3e455ef1ff90eafe10383fcd449289